tests/utils/test_config_utils.py

```diff
@@ -0,0 +1,170 @@
+    expected_config = None
+    assert expected_config == config_utils.get_hf_peft_config("", None, "")
+
+
+def test_get_hf_peft_config_returns_Lora_config_correctly():
+    # Test that when a value is not defined, the default values are used
+    # Default values: r=8, lora_alpha=32, lora_dropout=0.05, target_modules=["q_proj", "v_proj"]
+    tuning_config = peft_config.LoraConfig(r=3, lora_alpha=3)
+
+    config = config_utils.get_hf_peft_config("CAUSAL_LM", tuning_config, "")
+    assert isinstance(config, LoraConfig)
+    assert config.task_type == "CAUSAL_LM"
+    assert config.r == 3
+    assert config.lora_alpha == 3
+    assert config.lora_dropout == 0.05  # default value from peft_config.LoraConfig
+    assert config.target_modules == {
+        "q_proj",
+        "v_proj",
+    }  # default value from peft_config.LoraConfig
+
+    # Test that when target_modules is ["all-linear"], we convert it to str type "all-linear"
+    tuning_config = peft_config.LoraConfig(r=234, target_modules=["all-linear"])
+
+    config = config_utils.get_hf_peft_config("CAUSAL_LM", tuning_config, "")
+    assert isinstance(config, LoraConfig)
+    assert config.r == 234
+    assert config.target_modules == "all-linear"
+    assert config.lora_dropout == 0.05  # default value from peft_config.LoraConfig
+
+
+def test_get_hf_peft_config_returns_PT_config_correctly():
+    # Test that the prompt tuning config is set properly for each field
+    # when a value is not defined, the default values are used
+    # Default values:
+    # prompt_tuning_init="TEXT",
+    # prompt_tuning_init_text="Classify if the tweet is a complaint or not:"
+    tuning_config = peft_config.PromptTuningConfig(num_virtual_tokens=12)
+
+    config = config_utils.get_hf_peft_config("CAUSAL_LM", tuning_config, "foo/bar/path")
+    assert isinstance(config, PromptTuningConfig)
+    assert config.task_type == "CAUSAL_LM"
+    assert config.prompt_tuning_init == "TEXT"
+    assert config.num_virtual_tokens == 12
+    assert (
+        config.prompt_tuning_init_text == "Classify if the tweet is a complaint or not:"
+    )
+    assert config.tokenizer_name_or_path == "foo/bar/path"
+
+    # Test that tokenizer path is allowed to be None only when prompt_tuning_init is not TEXT
+    tuning_config = peft_config.PromptTuningConfig(prompt_tuning_init="RANDOM")
+    config = config_utils.get_hf_peft_config(None, tuning_config, None)
+    assert isinstance(config, PromptTuningConfig)
+    assert config.tokenizer_name_or_path is None
+
+    tuning_config = peft_config.PromptTuningConfig(prompt_tuning_init="TEXT")
+    with pytest.raises(ValueError) as err:
+        config_utils.get_hf_peft_config(None, tuning_config, None)
+        assert "tokenizer_name_or_path can't be None" in err.value
+
+
+def test_create_tuning_config():
+    # Test that LoraConfig is created for peft_method Lora
+    # and fields are set properly
+    tune_config = config_utils.create_tuning_config("lora", foo="x", r=234)
```

Comment:
But I don't think you have to keep sending in foo and testing it for each case. Since you test this in the `update_config()` which is waht is called in this method, that's probably suffificient